### PR TITLE
LoginController only for login flow

### PR DESCRIPTION
**Pull request by slaveek**
This pull request contains changes related to #549 issue.
Sorry for a small typo in branch name ;)

### CODE REVIEW

**Comment by panique** [comment]
Wow, thanks, this looks really good! :) Please gimme some days for testing!

**Comment by slaveek** [comment]
No problem! ;)

File: application/controller/RegisterController.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        } else {
+            $this->View->render('register/index');
+        }
+    }
+
+    /**
+     * Register page action
+     * POST-request after form submit
+     */
+    public function register_action()
```
Comment (geozak) [inline application/controller/RegisterController.php]: Wouldn't this be better as index_action() to fit the naming pattern used for action-action_post_handler theme used throughout this framework.

Corresponding change would be needed here: https://github.com/panique/huge/pull/747/files?diff=unified#diff-5ac2d5faa50e65202dd471632e2632f2R11

**Comment by geozak** [comment]
# Major changes missed:
1. application/config/config.development.php L149: Need to change "login/verify" to "register/verify".

# Minor changes missed:
1. (application/view/login/showProfile.php ? application/view/user/index.php) L2: Forgot to change "LoginController/showProfile" to "UserController/index".
2. (application/view/login/editUsername.php ? application/view/user/editUsername.php) L2: Forgot to change "LoginController/editUsername" to "UserController/editUsername".
3. README.md L378: Need to change "application/view/login/editUsername.php" to "application/view/user/editUsername.php"
4. (application/view/login/editUserEmail.php → application/view/user/editUserEmail.php) L2: Forgot to change "LoginController/editUserEmail" to "UserController/editUserEmail"
5. (application/view/login/changeUserRole.php → application/view/user/changeUserRole.php) L2: Forgot to change "LoginController/changeUserRole" to "UserController/changeUserRole"
6. README.md L363: Need to change "/login/changeUserRole" to "/user/changeUserRole"
7. application/view/index/index.php L10: Need to change "/login/register" to "/register/index" or other example of choice.
8. (application/view/login/changePassword.php → application/view/user/changePassword.php) L2: Forgot to change "LoginController/changePassword" to "UserController/changePassword"
9. (application/view/login/register.php → application/view/register/index.php) L33: Forgot to change "YOURURL/login/showcaptcha" to "YOURURL/register/showcaptcha"

# Potential changes to discuss:
1. Where should the reset password functionality exist? Login, Register, User, or elsewhere.
2. In application/view/_templates/header.php should the list items from line 51 to 68 be using checkForActiveControllerAndAction instead of checkForActiveController? 

# Summary:
There are a few small quickly fixable errors that I have listed above but otherwise this pull request is solid.
I have reviewed every changed line in the in the files changed section. And every method that was deleted has a corresponding addition.
I searched the whole repository, and every code reference to a deleted/moved line/method/file has been updated except for the instances I mentioned above.
None of these changes change how things flow further than the indented of breaking up the login, user, and register components into their own controls.

I hope this analysis helps get this request accepted faster.

**Comment by panique** [comment]
Thanks, I'll have a look on this on the weekend

**Comment by slaveek** [comment]
**Major and Minor changes missed**
Agree!
All thinks that @geozak described MUST be definitely fixed.

**Potential changes to discuss**
I'd like extend @geozak description:
3. Where should the Captcha functionality exist? Register or it's own controller or in core directory?

Captcha can be used in other functionalities not register user only.

**Comment by geozak** [comment]
@slaveek The captcha functionality is probably okay in the register controller because captchas as I've seen captcha have only been used in while registering user accounts.


I made the major and minor on a clean branch that I pulled slaveek/huge:login-cotroller into but I don't see slaveek/huge:login-cotroller on the list of forks to make to make a pull request to.
Here is a link to the branch https://github.com/geozak/huge/tree/login-controller-fixes
I didn't want to submit it staight to panique/huge until this main one get resolved.

**Comment by slaveek** [comment]
OK! I'll fix it and I'll add more commits in this branch.

**Comment by slaveek** [comment]
I'm SORRY. It's big mistake. It's call class Login which is doesn't exist. Sorry once again! I'll roll back the previous version.

**Comment by panique** [comment]
Thanks a lot, looks really good! :)

**Pull request merged**